mobify/commercecloud-ocapi-client
# Fix significantly outdated dependencies and security vulnerabilities

# Description

This PR includes updates to several outdated dependencies (including a few w significant security vulnerabilities in the old versions).

Fixes #19 

## Type of change


## Changes

.babelrc, rollup.config.js and package.json script updates related the following package updates.

_Minor versions updates:_
"atob": "^2.0.3">"^2.1.2"
"btoa": "^1.1.2">"^1.2.1"
"eslint-import-resolver-webpack": "^0.8.3">"eslint-import-resolver-webpack": "^0.11.1"
"eslint-plugin-import": "2.3.0">"eslint-plugin-import": "^2.17.2"
"express": "^4.16.2">"express": "^4.17.0"
"jsdoc": "^3.4.3">"jsdoc": "^3.6.2"
"mobify-code-style": "2.8.1">"mobify-code-style": "^2.8.4"

_Major version updates:_
"superagent": "3.5.2">"^5.0.5"
"babel-cli": "^6.24.1">"@babel/cli": "^7.4.4"
"babel-core": "6.18.0">"@babel/core": "^7.4.5"
"babel-preset-env": "^1.6.1">"@babel/preset-env": "^7.4.5"
"eslint": "3.19.0">"eslint": "^5.16.0"
"mocha": "~2.3.4">"mocha": "^6.1.4"
"rollup": "^0.49.3">"rollup": "^1.12.3"
"rollup-plugin-babel": "^3.0.2">"rollup-plugin-babel": "^4.3.2"
"rollup-plugin-commonjs": "^8.2.6">"rollup-plugin-commonjs": "^10.0.0"
"rollup-plugin-node-resolve": "^3.0.0">"rollup-plugin-node-resolve": "^5.0.0"
"sinon": "1.17.3">"sinon": "^7.3.2"

_Remove dependencies (should no longer be needed):_
"babel-plugin-external-helpers": "^6.22.0"

_Additional dependencies (required as peer for updated packages above - or for testing):_
"@babel/register": "^7.4.4"
"webpack": "^4.32.2"

# How to test this PR?

Ran full test suite using `npm run test` (including updating from test config in package.json from babel-core/register to @babel/register)

**IMPORTANT**: The following 2 existing tests failed due to coding issues in existing test (CustomersApi.spec.js) that are now flagged by the newer/updated version of "mocha" (Error: Resolution method is overspecified. Specify a callback *or* return a Promise). I have gone ahead and fixed these two issues but would love an extra pair of eyes on them as they are the only changes I needed to make to the tests in order to pass.
FAILED:
![Screen Shot 2019-05-23 at 11 29 57 AM](https://user-images.githubusercontent.com/160647/58270359-32d1a800-7d4f-11e9-86d4-353b196af99c.png)
UPDATED/PASSED:
![Screen Shot 2019-05-23 at 11 47 29 AM](https://user-images.githubusercontent.com/160647/58270975-9f997200-7d50-11e9-83d1-9ba1cca306e7.png)

NOTE: there are also few existing tests that throw an UnhandledPromiseRejectionWarning and should be addressed in the core code, but they appeared in existing test suite run before making any changes 
![Screen Shot 2019-05-23 at 11 44 15 AM](https://user-images.githubusercontent.com/160647/58271013-b2ac4200-7d50-11e9-97a2-100040f6ee9e.png)

# Checklist:

- [x] My code follows the style guidelines of this project (`npm run lint`)
- [x] New and existing unit tests pass locally with my changes (`npm test`)
